pkg/synopsysctl/cmd_serve_UI.go

```diff
@@ -0,0 +1,652 @@
+	// provided with the syopsysctl binary using the packr cli
+	box := packr.NewBox("../operator-ui-ember/dist")
+
+	// Create a Router to listen and serve User Interface requests
+	router := mux.NewRouter()
+
+	// // TODO: add operator for Black Duck for 12.0 release
+	// // api route - deploy_operator deploys Synopsys Operator into the cluster
+	// router.HandleFunc("/api/deploy_operator", func(w http.ResponseWriter, r *http.Request) {
+	// 	log.Infof("handling request: %q\n", html.EscapeString(r.URL.Path))
+	// 	reqBody, err := ioutil.ReadAll(r.Body)
+	// 	if err != nil {
+	// 		log.Fatal(err)
+	// 	}
+	// 	log.Debugf("request data: %s", reqBody)
+	// 	err = deployOperatorRequest(reqBody)
+	// 	if err != nil {
+	// 		log.Errorf("error deploying Synopsys Operator: %s\n", err)
+	// 		return
+	// 	}
+	// 	log.Infof("successfully deployed Synopsys Operator")
+	// })
+
+	// api route - deploy_polaris deploys Polaris into the cluster
+	router.HandleFunc("/api/ensure_polaris", func(w http.ResponseWriter, r *http.Request) {
+		log.Infof("handling request: %q\n", html.EscapeString(r.URL.Path))
+		reqBody, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		pConfig := PolarisUIRequestResponse{}
+		err = json.Unmarshal(reqBody, &pConfig)
+		if err != nil {
+			log.Errorf("failed to unmarshal Polaris request: %s", err)
+			return
+		}
+		err = checkRequiredPolarisRequestFields(pConfig)
+		if err != nil {
+			log.Errorf("missing data to ensure Polaris: %s", err)
+			return
+		}
+
+		polarisObj, err := convertPolarisUIResponseToPolarisObject(pConfig)
+		if err != nil {
+			log.Errorf("failed to convert Request data to Polaris Object: %s", err)
+			return
+		}
+
+		// Check Secret to see if polaris already exists
+		namespace = polarisObj.Namespace // set global namespace value for getPolarisFromSecret and ensurePolaris
+		oldPolaris, err := getPolarisFromSecret()
+		if err != nil {
+			log.Errorf("failed to get Secret for Polaris: %s", err)
+			return
+		}
+		if oldPolaris == nil { // create new Polaris
+			err = ensurePolaris(polarisObj, false)
+			if err != nil {
+				log.Errorf("error ensuring Polaris: %s", err)
+				return
+			}
+		} else { // update Polaris
+			err = ensurePolaris(polarisObj, true)
+			if err != nil {
+				log.Errorf("error ensuring Polaris: %s", err)
+				return
+			}
+		}
+
+		log.Infof("successfully ensured Polaris")
+	})
+
+	// api route - get_polaris returns Polaris specification to the client
+	router.HandleFunc("/api/get_polaris", func(w http.ResponseWriter, r *http.Request) {
+		log.Infof("handling request: %q\n", html.EscapeString(r.URL.Path))
+		reqBody, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// Get Polaris from secret
+		namespace = string(reqBody)
+		polarisObj, err := getPolarisFromSecret()
+		if err != nil {
+			log.Errorf("error getting Polaris: %s", err)
+			emptyResonse, _ := json.Marshal(PolarisUIRequestResponse{})
```

Comment:
maybe returning 404 not found is a better approach?....the contract being back-end and front-end is usually api status codes